apps/api/forms.py

```diff
@@ -14,9 +17,65 @@
 def choices(x):
     return [(c, c) for c in x]
 
+HASH_RE = re.compile(r'^[0-9a-f]{64}$', re.I)
 
-class PerformanceForm(happyforms.Form):
 
+class ChecksumsForm(happyforms.Form):
+    checksum_json = forms.CharField(max_length=2 ** 24 - 1,  # MEDIUMTEXT max.
+                                    required=True)
+
+    def clean_checksum_json(self):
+        # Do basic schema validation.
```

Comment:
Maybe... It's a field cleaning function, so its purpose is fairly clear. I'm not sure there's any benefit to exposing this particular detail in a docstring.